cluster/examples/kubernetes/ceph/cluster-stretched-aws.yaml

```diff
@@ -0,0 +1,136 @@
+#################################################################################################################
+# Define the settings for the rook-ceph cluster with common settings for a production cluster.
+# All nodes with available raw devices will be used for the Ceph cluster. At least three nodes are required
+# in this example. See the documentation for more details on storage settings available.
+
+# For example, to create the cluster:
+#   kubectl create -f crds.yaml -f common.yaml -f operator.yaml
+#   kubectl create -f cluster-stretched.yaml
+#################################################################################################################
+
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: rook-ceph
+  namespace: rook-ceph # namespace:cluster
+spec:
+  dataDirHostPath: /var/lib/rook
+  mon:
+    # Five mons must be created for stretch mode
+    count: 5
+    allowMultiplePerNode: false
+    stretchCluster:
+      # The ceph failure domain will be extracted from the label, which by default is the zone. The nodes running OSDs must have
+      # this label in order for the OSDs to be configured in the correct topology. For topology labels, see
+      # https://rook.github.io/docs/rook/master/ceph-cluster-crd.html#osd-topology.
+      failureDomainLabel: topology.kubernetes.io/zone
+      # The sub failure domain is the secondary level at which the data will be placed to maintain data durability and availability.
+      # The default is "host", which means that each OSD must be on a different node and you would need at least two nodes per zone.
+      # If the subFailureDomain is set to "osd", the OSDs would be allowed anywhere in the same zone including on the same node.
+      # If set to "rack" or some other intermediate failure domain, those labels would also need to be set on the nodes where
+      # the osds are started.
+      subFailureDomain: host
```

Comment:
Why are we doing OSD configuration under `mon.stretchCluster`?

Reply:
Fair question... There is a single CRUSH rule that is created for stretch clusters at the time the mons are created, which all pools must then use. The rule requires two levels which are represented here by the failureDomain and subFailureDomain. It made sense to keep them together IMO instead of splitting the subfailureDomain into the `storage` section where we don't have any stretch cluster settings. Perhaps `stretchCluster` should have been a top-level setting, but too late now...

Reply:
Why is it too late? This was experimental support before right? Why not treat that as a beta and change the config layout?

Reply:
The mon reconciler creates the CRUSH rule for the stretch cluster, so I don't think it's wrong that it's under the mon settings either, and doesn't seem necessary to change. These settings are already being used downstream, so if changed we would need to support the settings staying here for backward compatibility.

Reply:
Or we could make ocs-operator migrate the settings... It doesn't sit right with me, but I'm having trouble really weighing the options right now. Maybe having Seb's input would be useful.

Reply:
My view is still that it is valid under the `mon` section. I can see the argument that it's better in its own top-level stretch section, but I don't see a benefit to changing it given the backward compatibility costs if nothing else. It certainly is orthogonal to this PR if you want to open an issue.

Reply:
If this means a lot of trouble for the downstream implementation I'd keep it this way, even though I agree it should have been its own section (something to think about for the future I guess 🙂 ). Also using `mon` section is not entirely wrong but not entirely true either 😅